### DISCUSSION TITLE
pac power-fx command defaults to Default Environment

### DISCUSSION THREAD

**Discussion by jarioksanen** [microsoft/powerplatform-build-tools#610] [Q&A]
Hello,
Even if I'm logged in a another environment, the pac power-fx repl and run commands connects me directly to Default-Personal productive environment.
The power-fx command does not accept --environment parameter, so I cannot switch over to any other but the Default-Personal env.

**Comment by snizar007** [comment]
Hello @jarioksanen - Before using 'pac power-fx repl', please use 'pac org who' to know which environment you are connected to. 
Ensure you are using UNIVERSAL type environment and reply.

Thank you!

**Reply by jarioksanen** [reply]
Thanks @snizar007 for you response.
It still does not work in standalone terminal, but at least VS Code extension logs me in so that I can try to execute commands. Unfortunately, querying any of the Dataverse tales result nothing. For instance, a simple core table fetch brings nothing ("Accounts", "Contacts", ). And there are rows in the tables in the environment.

>> Filter(Accounts, "Christian" in name) 
Error 7-15: Name isn't valid. 'Accounts' isn't recognized.
Error 0-37: The function 'Filter' has some invalid arguments.
Error 32-36: Name isn't valid. 'name' isn't recognized.

**Comment by temmyraharjo** [comment]
Hi @snizar007,

I saw in the https://devblogs.microsoft.com/powerplatform/power-platform-command-line-interface-september-update/,
Seems like we need to run `pac power-fx repl`. Then the program will start running and we can put the PowerFX command to get the result.

But when I'm using the latest version, I've got the below result (which does not have ">>" and the terminal stuck). Is there any step that I'm missing?

![image](https://github.com/microsoft/powerplatform-build-tools/assets/8713243/b7fbede0-b4e2-4fd3-96e0-6760d6d26b86)

Additional information.
Using `pac power-fx run` also still gets the same error:
![image](https://github.com/microsoft/powerplatform-build-tools/assets/8713243/ffcc005c-3711-4802-9760-61a44e005c5a)

**Reply by snizar007** [reply]
Hello @temmyraharjo - 
1. If you want to run the Power Fx REPL, just type `pac power-fx repl` to get into the shell. Of course, you must be connected to the Power Platform environment, which you can check using `pac org who`. You can then type `First(Contacts)` to get the first record.
2. If you need to run Power Fx script (with Power Fx formulas), you can simply typ `pac power-fx run --file <filename>`. Make sure file name is without .\ like in PowerShell.

**Reply by temmyraharjo** [reply]
1. Yes, as in the first screenshot. You can see that I'm connected to the org. The issue is the same with @brasov2de where if I'm using the terminal (outside VS Code), it will not work (again, what Diana said is correct. If using Terminal in VS Code it can work as expected). I'm installing PAC CLI using MSI.
2. When running `pac power-fx run`, I encountered this error:
![image](https://github.com/microsoft/powerplatform-build-tools/assets/8713243/b6d81335-a65a-46ce-8318-234c7a24b606)

For point 2, do I need to install anything? Why the system don't know the "Contacts" table and return the First record?

**Comment by brasov2de** [comment]
I have the same issue as @temmyraharjo  inside the standalone Powershell Terminal. It works only inside the VSCode Terminal.

**Reply by snizar007** [reply]
Did you install PAC CLI using `dotnet tool` or with MSI outside of VSCode?
If you used `dotnet tool` to install, it should work.

**Reply by jarioksanen** [reply]
I have installed with dotnet, but face the same issue still.

**Reply by brasov2de** [reply]
I've installed it with msi. It's installed from the old days, when there was no VSCode extension. I still keep it up to date, and try some commands there. There weren't any issues until now.

**Comment by jarioksanen** [comment]
UPDATE: For me both issues with pac power-fx (defaults to "Default-Personal productive environment" and pac repl commands not recognizing DV tables) got fixed when I removed all current environments from the auth list, and added only the one I wanted to use as the default environment.

**Reply by jarioksanen** [reply]
UPDATE2: The authentication profile must be UNIVERSAL, not DATAVERSE.

**Reply by temmyraharjo** [reply]
I tested it on my side and it still not works. The "--kind" parameter is deprecated and when I'm using it, I found the URL store in "pac org who" is globaldisco and not specified to the URL I set. 😭 And when I tried to execute (as "UNIVERSAL"), all the commands got below error:
![image](https://github.com/microsoft/powerplatform-build-tools/assets/8713243/4b058928-9d3b-48e5-b7e8-f486a9e880b5)

**Comment by petrochuk** [comment]
We are looking at fixing these inconsistencies
